Plugin~/Src/MeshUtils/Include/MeshUtils/muMath.h

```diff
@@ -1651,4 +1651,12 @@ static inline void enumerate(A1& a1, A2& a2, A3& a3, const Body& body)
         body(a1[i], a2[i], a3[i]);
 }
 
+template
+<typename T>
+T roundToDecimals(T val, float decimalPlaces = 2) {
+    assert(decimalPlaces >= 1);
+    float roundingFactor = std::pow(10, decimalPlaces);
```

Comment:
using `double` and then returning as `T` may be more accurate ?

Reply:
This needs to work with types that cannot be cast to double like float3, etc. Also, this rounding function is only to be used for hashing, accuracy isn't really important. We just need to clamp floats that are very similar to the same number so their hashes are the same.